taxi/views.py

```diff
@@ -61,17 +63,40 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        car = self.get_object()
+        context["driver_ids"] = list(car.drivers.values_list("id", flat=True))
+
+        return context
+
+
+@login_required
+def toggle_driver(request: HttpRequest, pk: int) -> HttpResponse:
+    car = get_object_or_404(Car, pk=pk)
+
+    if request.user in car.drivers.all():
+        car.drivers.remove(request.user)
+    else:
+        car.drivers.add(request.user)
```

Comment:
1. you don't prefetch drivers
2. it's better to use .exists() method because in this case you fetch 1 000 000 records and then python loops through all objects to find if there is request.user. in case you use .exists(), db will check it using "where" condition